adapters/mock.go

```diff
@@ -31,6 +32,60 @@ func NewMockAdapter(isClient bool) *MockAdapter {
 
 var _ Adapter = (*MockAdapter)(nil)
 
+func (m *MockAdapter) Batch(ctx context.Context, kind domain.Kind, _ domain.BatchType, items domain.BatchItems) error {
```

Comment:
TODO this is the kind of logic I would like to move to core, but it needs a clever refactoring - once we settle down a bit